binary/function.proto

```diff
@@ -124,15 +124,26 @@ message FunctionSignature {
     }
 
     message Argument {
-        ParameterizedType type = 1;
-        ArgumentKind kind = 2;
-        string name = 3;
+        string name = 1;
```

Comment:
In the YAML it looks like options/optional are part of the "parameters" section but here it looks like you are defining them for the "arguments" section?

Reply:
Yeah, i think this is clearer/more natural this way when using protobuf but I think it is easier to read/consume in yaml the other way (since there are no message types in Yaml). Basically, trying to be idiomatic to each form.